src/graphics/aurora/borderquad.cpp

```diff
@@ -0,0 +1,188 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file
+ * A quad for generating borders
+ */
+
+#include <cfloat>
+
+#include "borderquad.h"
+#include "textureman.h"
+
+namespace Graphics {
+
+namespace Aurora {
+
+BorderQuad::BorderQuad(Common::UString edge, Common::UString corner, float x, float y, float w, float h)
+		: Renderable(kRenderableTypeGUIFront),
+          _edgeWidth(0), _edgeHeight(0), _cornerWidth(0), _cornerHeight(0),
+          _x(x), _y(y), _w(w), _h(h) {
+	_edge = TextureMan.get(edge);
+	_corner = TextureMan.get(corner);
+
+    _cornerWidth = _corner.getTexture().getWidth();
+    _cornerHeight = _corner.getTexture().getHeight();
+    _edgeWidth = _edge.getTexture().getWidth();
+    _edgeHeight = _edge.getTexture().getHeight();
+
+    _distance = -FLT_MAX;
+
+    assert(!_corner.empty());
+    assert(!_edge.empty());
+}
+
+void BorderQuad::setPosition(float x, float y, float UNUSED(z)) {
+    lockFrameIfVisible();
+
+    _x = (int)x;
+    _y = (int)y;
```

Comment:
Can you leave a comment why you are casting to int? Also, why not use static_cast<>?